extensions/notebook/src/dialog/remoteBookDialog.ts

```diff
@@ -0,0 +1,249 @@
+import { RemoteBookController, IReleases } from '../book/remoteBookController';
+
+function getRemoteLocationCategory(name: string): azdata.CategoryValue {
+	if (name === 'GitHub') {
+		return { name: name, displayName: loc.onGitHub };
+	}
+	return { name: name, displayName: loc.onSharedFile };
+}
+
+export class RemoteBookDialogModel {
+	private _remoteTypes: azdata.CategoryValue[];
+	private _controller: RemoteBookController;
+
+	constructor(
+	) {
+		this._controller = new RemoteBookController();
+	}
+
+	public get remoteLocationCategories(): azdata.CategoryValue[] {
+		if (!this._remoteTypes) {
+			this._remoteTypes = [getRemoteLocationCategory('GitHub'), getRemoteLocationCategory('Shared File')];
+		}
+		return this._remoteTypes;
+	}
+
+	public async getReleases(url: string): Promise<IReleases[]> {
+		let remotePath: URL;
+		try {
+			url = 'https://api.github.com/'.concat(url);
+			remotePath = new URL(url);
+			return this._controller.getReleases(remotePath);
+		}
+		catch (error) {
+			throw error;
+		}
+	}
+
+	public async downloadLocalCopy(url: URL, remoteLocation: string, release?: IReleases): Promise<string[]> {
+		if (release) {
+			return await this._controller.setRemoteBook(url, remoteLocation, release.zipURL, release.tarURL);
+		}
+		else {
+			return await this._controller.setRemoteBook(url, remoteLocation);
+		}
+	}
+
+	public openRemoteBook(book: string): void {
+		this._controller.openRemoteBook(book);
+	}
+
+}
+
+export class RemoteBookDialog {
+
+	private dialog: azdata.window.Dialog;
+	private view: azdata.ModelView;
+	private formModel: azdata.FormContainer;
+	private urlInputBox: azdata.InputBoxComponent;
+	private remoteLocationDropdown: azdata.DropDownComponent;
+	private remoteBookDropdown: azdata.DropDownComponent;
+	private releaseDropdown: azdata.DropDownComponent;
+	private releases: IReleases[];
+
+	constructor(private model: RemoteBookDialogModel) {
+	}
+
+	public showDialog(): void {
+		this.createDialog();
+	}
+
+	private createDialog(): void {
+		this.dialog = azdata.window.createModelViewDialog(loc.openRemoteBook);
+		this.dialog.registerContent(async view => {
+			this.view = view;
+
+			this.remoteLocationDropdown = this.view.modelBuilder.dropDown().withProperties({
+				values: this.model.remoteLocationCategories,
+				value: '',
+				editable: false,
+			}).component();
+
+			this.remoteLocationDropdown.onValueChanged(e => this.onRemoteLocationChanged());
+
+			this.urlInputBox = this.view.modelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({
+					placeHolder: loc.url
+				}).component();
+
+			this.urlInputBox.onTextChanged(async () => await this.validate());
+
+			this.remoteBookDropdown = this.view.modelBuilder.dropDown().withProperties({
+				values: [],
+				value: '',
+			}).component();
+
+			this.releaseDropdown = this.view.modelBuilder.dropDown().withProperties({
+				values: [],
+				value: '',
+			}).component();
+
+			this.releaseDropdown.updateCssStyles({
```

Comment:
You can specify the property during creation instead of calling it after the fact. 

(you also have this code duplicated)